source/AI.cpp

```diff
@@ -119,10 +119,74 @@ namespace {
 	void Deploy(const Ship &ship, bool includingDamaged)
 	{
 		for(const Ship::Bay &bay : ship.Bays())
-			if(bay.ship && (includingDamaged || bay.ship->Health() > .75))
+			if(bay.ship && (includingDamaged || bay.ship->Health() > .75) &&
+					(!bay.ship->IsYours() || bay.ship->IsDeploying()))
 				bay.ship->SetCommands(Command::DEPLOY);
 	}
 	
+	// Issue deploy orders for the selected ships (or the full fleet if no ships are selected).
+	void IssueDeploy(const PlayerInfo &player)
+	{
+		// Figure out what carried ships we are giving orders to
+		vector<Ship *> ships;
+		bool fullFleet = player.SelectedShips().empty();
+		if(fullFleet)
+		{
+			for(const shared_ptr<Ship> &it : player.Ships())
+				if(it.get() != player.Flagship() && !it->IsParked() && it->CanBeCarried())
+					ships.push_back(it.get());
+		}
+		else
+			for(const weak_ptr<Ship> &it : player.SelectedShips())
+			{
+				shared_ptr<Ship> ship = it.lock();
+				if(ship)
+					ships.push_back(ship.get());
```

Comment:
Is this missing a "CanBeCarried" or "HasBays" condition?

Also note that player ships are not [deleted until the player lands](https://github.com/endless-sky/endless-sky/blob/d3ccbdd72f71b59b6539dc5658b8d201ae2cdf0a/source/PlayerInfo.cpp#L1088-L1105), so if a selected ship gets destroyed, it will still be present in `SelectedShips()` until the player re-selects other ships. Note that this means that `SelectedShips()` can possibly contain only destroyed ships, in which case we probably want to treat pressing the deploy/recall key as the "full fleet" scenario.

Reply:
Agreed, it was missing CanBeCarried, and it was also missing IsDestroyed. I added both and changed the order to first go through selected ships.